bundles/org.openhab.core.model.script/src/org/openhab/core/model/script/internal/actions/TimerImpl.java

```diff
@@ -48,26 +46,29 @@ public TimerImpl(Scheduler scheduler, ZonedDateTime startTime, SchedulerRunnable
 
     @Override
     public boolean cancel() {
-        cancelled = true;
         return future.cancel(true);
     }
 
     @Override
     public boolean reschedule(ZonedDateTime newTime) {
```

Comment:
Not related to your contribution but I am wondering if we should synchronize this method to have a thread safe implementation? Wdyt?

Reply:
It appears to be thread safe already https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/CompletableFuture.html#cancel(boolean)

Reply:
What I am talking about is a small race condition on the `future` property of this implementation when calling the `reschedule(...)` method because we first cancel the existing future and afterwards assign a new `CompletableFuture` to it.